File: src/main/java/com/selqet/momrecipebook/entity/IngredientEntity.java
package com.selqet.momrecipebook.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(schema = "public", name = "ingredient")
public class IngredientEntity {

    //TODO Read about columnDefinition
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "ingredient_generator")
    @SequenceGenerator(name = "ingredient_generator", sequenceName = "ingredient_ingr_id_seq", allocationSize = 1)
    @Column(name = "ingr_id")
    private Long id;

    @Column(name = "ingr_name")
    private String title;

    @Column(name = "calories")
    private Short calories;
}

File: src/main/kotlin/com/example/momrecipebook/model/IngredientEntity.kt
package com.example.momrecipebook.model

import javax.persistence.*

@Entity
@Table(name = "ingredient")
data class IngredientEntity(
    @Id
    @SequenceGenerator(name="ingredient_ingr_id_seq",
        sequenceName="ingredient_ingr_id_seq",
        allocationSize=1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE,
        generator="ingredient_ingr_id_seq")
    @Column(name = "ingr_id", updatable = false)
    var id: Long,

    @Column(name = "ingr_name")
    var name: String,

    @Column(name = "calories")
    var calories: Short?
)
